cEP-0016.md

```diff
@@ -0,0 +1,332 @@
+
+```
+[commits]
+bears = GitCommitBear
+shortlog_regex = ([^:]*|[^:]+: [A-Z0-9*].*)
+shortlog_trailing_period = False
+```
+
+Projects- [coala/coala-bears](https://github.com/coala/coala-bears), [coala/coala-quickstart](https://github.com/coala/coala-quickstart), [coala/corobo](https://github.com/coala/corobo)
+
+```
+[all.commit]
+bears = GitCommitBear
+shortlog_trailing_period = False
+shortlog_regex = ([^:]*|[^:]+: [A-Z0-9*].*)
+body_close_issue = True
+body_close_issue_full_url = True
+body_close_issue_on_last_line = True
+body_enforce_issue_reference = False
+```
+
+Project- [coala/coala-atom](https://github.com/coala/coala-atom)
+
+```
+[commit]
+bears = GitCommitBear
+```
+
+## Proposed Change
+
+This project would allow organizations to customize the `.coafile`
+to allow following settings for `GitCommitBear`.
+
+```
+[commit]
+bears = GitCommitBear, VCSCommitBear, SkipCIBear, MergeCommitBear, RevertCommitBear
+allow_skip_ci_build = False
+allow_git_merge_commit = False
+allow_git_revert_commit = True
+```
+
+GitHub Pull Request temporary merge commits commits would be inspected by
+making changes in the `GitCommitBear`. For other special commits, a meta bear
+called `VCSCommitBear` would be used to provide result of the type of commit
+which will then be used for inspection in separate bears.
+
+Here is the detailed implementation stepwise:
+
+1. We start by identifying different types of special commits namely GitHub Pull
+   Request temporary merge commits, `git revert` , `git merge` and commits that
+   disable CI builds.
+2. Each type of commit is then handled as a separate case.
+3. For GitHub Pull Request temporary merge commits:
+   - As a solution to this problem, settings will be added in `GitCommitBear`
+     that would identify GitHub Pull Request temporary merge commits and ignore
+     them.
+   - `GitCommitBear` would identify and check contents of the unmerged parent of
+     these merge commits.
+4. A meta bear called `VCSCommitBear` would be created that would return a
+   Hidden Result comprising of a dictionary that will contain information about
+   the type of git commit, commit sha and a list of modified files.
+   This meta bear would be used to inspect each type of special commit.
+5. For commits that disable CI builds:
+   This would involve making a new bear named `SkipCIBear` to inspect commits
+   that disable CI builds. This bear would contain metadata about various CI
+   providers that are supported by coala such as Circle CI, Travis CI,
+   Semaphore CI etc. Most of the CI providers use sequences like `[ci skip]` and
+   `[skip ci]` for skipping CI build on commits. Commits that have either of
+   these sequences anywhere in the commit message are ignored by them. This
+   bear would detect the changes in such commits and verify that they can skip
```

Comment:
a critical aspect of this bear is that while all CI might skip the commit, GitMate will not skip it, and it will block the merge.

Reply:
So what should I do about it? Does this need to handled as a special case ?

Reply:
Open an issue in https://gitlab.com/gitmate/open-source/gitmate-2
And lets discuss about that

Reply:
Done https://gitlab.com/gitmate/open-source/gitmate-2/issues/382